### PR TITLE
Canary Radio BSP

### PR DESCRIPTION
**Pull request by lolsborn**
## Thank you for sending in a pull request, here's some tips to get started!

(Please delete all these tips and replace with your text)

- Before starting on some new big chunk of code, it it is optional but highly recommended to open an issue first
  to say "hey, I think this idea X should be implemented and I'm starting work on it. My general plan is Y, any feedback
  is appreciated." This will allow other devs to potentially save you time by not accidentially duplicating work etc...
- Please do not check in files that don't have real changes
- Please do not reformat lines that you didn't have to change the code on
- We recommend using the [Visual Studio Code](https://platformio.org/install/ide?install=vscode) editor along with the ['Trunk Check' extension](https://marketplace.visualstudio.com/items?itemName=trunk.io) (WSL2 is required on windows),
  because it automatically follows our indentation rules and its auto reformatting will not cause spurious changes to lines.
- If your PR fixes a bug, mention "fixes #bugnum" somewhere in your pull request description.
- If your other co-developers have comments on your PR please tweak as needed.
- Please also enable "Allow edits by maintainers".

### CODE REVIEW

File: bin/check-all.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ if [[ $# -gt 0 ]]; then
     # can override which environment by passing arg
     BOARDS="$@"
 else
-    BOARDS="tlora-v2 tlora-v1 tlora_v1_3 tlora-v2-1-1.6 tbeam heltec-v1 heltec-v2.0 heltec-v2.1 tbeam0.7 meshtastic-diy-v1 rak4631 rak4631_eink rak11200 t-echo pca10059_diy_eink"
+    BOARDS="tlora-v2 tlora-v1 tlora_v1_3 tlora-v2-1-1.6 tbeam heltec-v1 heltec-v2.0 heltec-v2.1 tbeam0.7 meshtastic-diy-v1 rak4631 rak4631_eink rak11200 t-echo canary pca10059_diy_eink"
```
Comment (hdngr) [inline bin/check-all.sh]: should we version this?  E.g. `canary-v1`?

File: boards/canary.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      "maximum_size": 815104,
+      "speed": 115200,
+      "protocol": "jlink",
+      "protocols": ["jlink", "nrfjprog", "nrfutil", "stlink"],
+      "use_1200bps_touch": true,
+      "require_upload_port": true,
+      "wait_for_upload_port": true
+    },
+    "url": "https://canaryradio.io/",
+    "vendor": "Canary IO, LLC"
```
Comment (hdngr) [inline boards/canary.json]: Canary Radio Company

File: src/mesh/generated/meshtastic/mesh.pb.h [hunk 3]
```diff
@@ -59,6 +59,8 @@ typedef enum _meshtastic_HardwareModel {
     meshtastic_HardwareModel_TLORA_T3_S3 = 16,
     /* B&Q Consulting Nano G1 Explorer: https://wiki.uniteng.com/en/meshtastic/nano-g1-explorer */
     meshtastic_HardwareModel_NANO_G1_EXPLORER = 17,
+    /* CanaryRadio.io LLC Handheld https://canaryradio.io */
```
Comment (hdngr) [inline src/mesh/generated/meshtastic/mesh.pb.h:66]: Canary Radio Company

Comment (lolsborn) [reply src/mesh/generated/meshtastic/mesh.pb.h:66]: This is a generated file and will be fixed in the protobuf repository

**Comment by hdngr** [comment]
@lolsborn closing in favor of https://github.com/meshtastic/firmware/pull/3150

**Pull request closed without merging**